pynamodb/models.py

```diff
@@ -441,11 +441,22 @@ def update(self, actions: List[Action], condition: Optional[Condition] = None, *
         self.deserialize(item_data)
         return data
 
-    def save(self, condition: Optional[Condition] = None, *, add_version_condition: bool = True) -> Dict[str, Any]:
+    def save(self, condition: Optional[Condition] = None, *, add_version_condition: bool = True, return_values_on_condition_failure: Optional[str] = None) -> Dict[str, Any]:
```

Comment:
This could be typed as:

```suggestion
    def save(self, condition: Optional[Condition] = None, *, add_version_condition: bool = True, return_values_on_condition_failure: Optional[Literal["ALL_OLD", "NONE"]] = None) -> Dict[str, Any]:
```